include/circt/Dialect/FIRRTL/FIRAnnotations.h

```diff
@@ -0,0 +1,31 @@
+//===- FIRAnnotations.h - .fir to FIRRTL dialect parser -------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Provide utilities related to dealing with Scala FIRRTL Compiler Annotations.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_FIRRTL_FIRANNOTATIONS_H
+#define CIRCT_DIALECT_FIRRTL_FIRANNOTATIONS_H
+
+#include "circt/Support/LLVM.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "llvm/ADT/StringMap.h"
+#include "llvm/Support/JSON.h"
```

Comment:
Please forward declare things in headers to reduce transitive includes, e.g.:

`namespace llvm { namespace json { class Path; class Value; }}`
`namespace mlir { class ArrayAttr; }`

It looks like StringMap isn't in LLVM.h, so StringMap.h is ok.

Reply:
Fixed in rebase.